src/components/state-history-charts.ts

```diff
@@ -83,6 +85,13 @@ class StateHistoryCharts extends LitElement {
     `;
   }
 
+  protected shouldUpdate(changedProps: PropertyValues): boolean {
+    return (
+      hasConfigOrEntityChanged(this, changedProps) ||
```

Comment:
This one is for lovelace cards, and should not be used here. 

You should add `isLoadingData`.

Reply:
Do you mean only isLoadingData or also historyData

Reply:
both

Reply:
Done and CI passed